nixos/modules/services/hardware/thermald.nix

```diff
@@ -4,27 +4,392 @@ with lib;
+        description = ''
+          Cooling device can be specified by a type and optionally a sysfs path. If the type is
+          already present in thermal sysfs, there is no need of a path. Compensation can use
+          min/max and step  size  to  increasing cool the system. Debounce period can be used
+          to force a waiting period for action.
+        '';
+      };
+    };
+  };
+
+  thermalSensorConfig = with types; submodule {
+    options = {
+      Type = mkOption {
+        type = string;
+        description = "The type of this sensor";
+      };
+
+      Path = mkOption {
+        type = nullOr path;
+        default = null;
+        description = "The path to this sensor";
+        example = "/sys/class/thermal/thermal_zone1/";
+      };
+
+      AsyncCapable = mkOption {
+        type = nullOr bool;
+        default = null;
+        description = "if true then we don't need to poll";
+      };
+    };
+  };
+
+  thermalZoneConfig = with types; submodule {
+    options = {
+      Type = mkOption {
+        type = string;
+        description = "The type of thermal zone";
+      };
+
+      Path = mkOption {
+        type = nullOr path;
+        default = null;
+        description = "The path to this thermal zone device";
+      };
+
+      AsyncCapable = mkOption {
+        type = nullOr bool;
+        default = null;
+        description = "if true then we don't need to poll";
+      };
+
+      TripPoints = mkOption {
+        type = (listOf tripPointConfig);
+        description = "The trip points of this thermal zone";
+      };
+    };
+  };
+
+
+  tripPointConfig = with types; submodule {
+    options = {
+      SensorType = mkOption {
+        type = string;
+        description = "The sensor type";
+      };
+
+      Temperature = mkOption {
+        type = int;
+        description = "Temperature at which to take action";
+      };
+
+      Type = mkOption {
+        type = enum [ "max" "passive" "active" ];
+        description = ''
+          If a max type is specified, then daemon will use PID control to
+          aggressively throttle to avoid reaching this temp.
+        '';
+      };
+
+      ControlType = mkOption {
+        type = nullOr (enum [ "SEQUENTIAL" "PARALLEL" ]);
+        default = null;
+        description = ''
+          When a trip point temp is violated, then number of cooling devices can be activated.
+          If control type is SEQUENTIAL then, it will exhaust first cooling device before trying
+          next.
+        '';
+      };
+
+      CoolingDevices = mkOption {
+        type = (listOf tripPointCoolingDeviceConfig);
+        default = [];
+        description = ''
+          The cooling devices to use for this trip point
+        '';
+      };
+    };
+  };
+
+  tripPointCoolingDeviceConfig = with types; submodule {
+    options = {
+      Index = mkOption {
+        type = nullOr int;
+        default = null;
+        description = "The index of this cooling device";
```

Comment:
I _think_ this might be related to the index of the acpi device and so I'm not sure if we should infer it this way. 

In particular I'm thinking of it from the perspective of someone copying a config from the internet, I'd imagine they'd expect an `index` parameter and would be surprised to find it missing, so I feel like it might be worth keeping even though we _might_ be able to infer it.

What do you think?

Reply:
Hmm yeah, I'd still prefer not having unnecessary options, uniformity over NixOS options seems more important than keeping the original config file options.

Would be good to have more information on whether this index server another purpose than just telling the index in the list.